### ISSUE TITLE
Fix false positives for interpolations in CSS-in-JS

### ISSUE THREAD

**Issue by hudochenkov** [stylelint/stylelint#4953]
> Clearly describe the bug

False positives for interpolations in CSS-in-JS. I think problem with parser. It might be also the same problem as https://github.com/stylelint/stylelint/issues/4713

> Which rule, if any, is the bug related to?

Probably many rules, as it's more likely a parser problem. I found these so far:

- `font-family-name-quotes`
- `font-family-no-missing-generic-family-keyword`
- `value-keyword-case`

> What code is needed to reproduce the bug?

```js
import styled from 'styled-components';

export const Button = styled.button`
	font-size: ${(props) => props.theme.fontSizeBase};
	font-family: ${(props) => props.theme.fontStackBase};
	font: ${({ theme }) => `normal ${theme.fontSizeBase}/${theme.lineHeightBase} ${theme.fontStackBase}`};
`;
```

- [Demo for all rules](https://stylelint.io/demo#N4Igxg9gJgpiBcICWBbADhATgFwAQGdsBPAGxilwDNMIVcByQ08gWknQgDsZPt96A3AB1OImAA8MOXJE6FcAIQCu2bF1wBeAsTJQAdACMVazgAMRQ7JS7YW+JAC8Y8XABJgACjQ00+AJSaAHy43hC+etgAFjAoMHrWvADKjjAKAIb4MAC+wqJWNiyUaShIJEQu7l4+-kEh1RHRsfE2idhpYADW6Zk5Fvm8FZ7AuFExMLhZARrBppxYKGkkbsCjTQnYyU7d2QD07qtxJEjcABIwSADmkdjbWcsHzUltnbemvWa5IAA0IPhEvGlxAhwPh8CxjiwAFb4b7gLiUS7A4AiXC4IQgTBKMj4dEuZGcVGo9HrQrFUpEFicYowFgARyUEGwMBxCHRiwA7mkiGClJxsWCOjAiOysFB0V8UYTiQUiiUypSICwSqDjhcWBceDBMEgwKS5RTBcLRbiRpiYBKCVKQAA3RZKGmGkWYKBsDIwE3okgQdla9GSrIiLIgLJAA)
- [Demo for `font-family-name-quotes`](https://stylelint.io/demo#N4Igxg9gJgpiBcICWBbADhATgFwAQGdsBPAGxilwDNMIVcByQ08gWknQgDsZPt96A3AB1OImAA8MOXJE6FcAIQCu2bF1wBeAsTJQAdACMVazgAMRQ7JS7YWlAIYokJIvFwASYAAo0NNPgBKTQA+XF8Ifz1sAAsYFBg9a14AZWx7MABrBXt8GABfYVErGzdPL2BcGLiYXDygjWCLbEtTTiwUexIPYCr4xJtkpAAvGGzcvIB6T16EkiRuAAkYJABzaOwx-O6Z-pS0zM280wKRU0KQABoQfCJee3EEcHx8FnmWACt8S-AuSlXH4AiXC4IQgTBKMj4UFuQGcYHA0FJWwOJwuFicRwwFgARyUEGwMChCFBnQA7vYiC8lJxIS8MjAiKSsFBQUDaiI8iA8kA)
- [Demo for `font-family-no-missing-generic-family-keyword`](https://stylelint.io/demo#N4Igxg9gJgpiBcICWBbADhATgFwAQGdsBPAGxilwDNMIVcByQ08gWknQgDsZPt96A3AB1OImAA8MOXJE6FcAIQCu2bF1wBeAsTJQAdACMVazgAMRQ7JS7YWlAIYokJIvFwASYAAo0NNPgBKTQA+XF8Ifz1sAAsYFBg9a14AZWx7MABrBXt8GABfYTNCkAAaEHwiXntxBHB8fBYkThYAK3xS8C5KJABzWuARXFwhEEwlMnwRtwHOIaGRpNsHJxcWTggWJ3qmnpYenhhMJDA7R2ciFgyYIgB3LCgp3GwxmEHcPJE8kDygA)
- [Demo for `value-keyword-case`](https://stylelint.io/demo#N4Igxg9gJgpiBcICWBbADhATgFwAQGdsBPAGxilwDNMIVcByQ08gWknQgDsZPt96A3AB1OImAA8MOXJE6FcAIQCu2bF1wBeAsTJQAdACMVazgAMRQ7JS7YW+JAC8Y8XABJgACjQ00+AJSaAHy43hC+etgAFjAoMHrWvADKjjAKAIb4MAC+wma5IAA0IPhEvGniCOD4+CxInCwAVviF4FyUSADmlcAiuLhCIJhKZPgDLj2cfX0DAG5pJEowLADWMEQA7lhQbBkwY-0gJBDrMJgDvbhZIlkgWUA)

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "font-family-name-quotes":"always-unless-keyword",
    "font-family-no-missing-generic-family-keyword": true,
    "value-keyword-case": "lower"
  }
}
```

> Which version of stylelint are you using?

`13.7.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.js"`.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, interpolations in CSS-in-JS

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
5:25
error
Unexpected missing generic font family (font-family-no-missing-generic-family-keyword)

6:22
error
Unexpected missing generic font family (font-family-no-missing-generic-family-keyword)

5:25
error
Expected quotes around "=> props.theme.fontStackBase}" (font-family-name-quotes)

6:22
error
Expected quotes around "=> `normal" (font-family-name-quotes)

4:26
error
Expected "props.theme.fontSizeBase}" to be "props.theme.fontsizebase}" (value-keyword-case)
```

**Comment by jeddy3** [comment]
Would this also be address by https://github.com/stylelint/stylelint/pull/4944 (and the work upstream)? I think this is how the styled-components processor worked too.

**Comment by hudochenkov** [comment]
It might.

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

We'll also [be deprecating](https://github.com/stylelint/postcss-css-in-js/issues/225) our [forked CSS-in-JS package](https://github.com/stylelint/postcss-css-in-js), in favour of leaner custom syntaxes.

Please consider [writing a custom syntax](https://github.com/stylelint/stylelint/blob/main/docs/developer-guide/syntaxes.md) specific to the CSS-in-JS library in this issue and fixing the bug there.
